cpp/modmesh/buffer/SimpleArray.hpp

```diff
@@ -1162,13 +1172,30 @@ class SimpleArray
     {
         if (!m_shape.empty())
         {
-            m_buffer = buffer_type::construct(m_shape[0] * m_stride[0] * ITEMSIZE);
+            m_buffer = buffer_type::construct(m_shape[0] * m_stride[0] * ITEMSIZE, 0);
             m_body = m_buffer->template data<T>();
         }
     }
 
     // NOLINTNEXTLINE(modernize-pass-by-value)
-    explicit SimpleArray(small_vector<size_t> const & shape, value_type const & value)
+    SimpleArray(small_vector<size_t> const & shape, with_alignment_t, size_t alignment)
```

Comment:
Use  a `with_alignment_t` to distinguish the ctors.

Reply:
Tagging argument should be at the end.